emcc.py

```diff
@@ -3311,8 +3311,10 @@ def phase_binaryen(target, options, wasm_target):
     # the only reason we need intermediate debug info, we can stop keeping it
     if settings.ASYNCIFY:
       intermediate_debug_info -= 1
-    if intermediate_debug_info and should_run_binaryen_optimizer():
-      # See https://github.com/emscripten-core/emscripten/issues/15269
+    # currently binaryen's DWARF support will limit some optimizations; warn on
+    # that. see https://github.com/emscripten-core/emscripten/issues/15269
+    dwarf_info = settings.DEBUG_LEVEL >= 3
+    if dwarf_info:
```

Comment:
But wait, doesn't `intermediate_debug_info` prevent wasm-opt from optimizing fully?

Reply:
Actually you are right.. wams-opt is always run one more time to remove any intermediate-only debug info

Reply:
Yes, exactly. Also it will run `--strip-dwarf` early on, if we don't want dwarf in the final output,

https://github.com/emscripten-core/emscripten/blob/e1d97f55d12ed63dc30e547c818af9a63be0b976/tools/building.py#L1489-L1501